### PR TITLE
Add restore feature

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

File: file_syncer/syncer.py [hunk 1]
```diff
@@ -258,6 +280,33 @@ def _get_remote_files(self):
...
+            local_filename = local_filename[1:]
+
+        driver = self._get_driver_instance()
+        filepath = os.path.join(self._directory, local_filename)
+
+        try:
+            obj = driver.get_object(container_name=self._container_name,
+                                    object_name=name)
+        except ObjectDoesNotExistError:
+            self._logger.debug('Manifest doesn\'t exist, assuming that ' +
```
Comment (Kami) [inline file_syncer/syncer.py]: Need to update the log message to reflect an actual error.

File: file_syncer/syncer.py [hunk 2]
```diff
@@ -258,6 +280,33 @@ def _get_remote_files(self):
 
         return parsed
 
+
+    def _download_remote_file(self, name):
+        """
+        Return a list of files in a container.
+        """
+
+        self._logger.debug('Downloading object: %(name)s', {'name': name})
```
Comment (Kami) [inline file_syncer/syncer.py]: Would be nice to also include the destination path in the message.

File: file_syncer/syncer.py (hunk 1 above)
Comment (rphillips) [inline file_syncer/syncer.py]: fixed in f0d248b

File: file_syncer/syncer.py (hunk 2 above)
Comment (rphillips) [inline file_syncer/syncer.py]: fixed

**Comment by rphillips** [comment]
rebased the PR

**Comment by Kami** [comment]
LGTM, +1

**Comment by rphillips** [comment]
I can't merge it :)

**Comment by Kami** [comment]
Merged, thanks :)

**Comment by Kami** [comment]
FYI - New version with this functionality has been published to PyPi.

**Comment by rphillips** [comment]
Awesome! Thanks

**Pull request merged**